components/layout/flex.rs

```diff
@@ -939,13 +940,29 @@ impl Flow for FlexFlow {
     }
 
     fn assign_block_size(&mut self, layout_context: &LayoutContext) {
-        self.block_flow
-            .assign_block_size_block_base(layout_context,
-                                          None,
-                                          MarginsMayCollapseFlag::MarginsMayNotCollapse);
         match self.main_mode {
-            Direction::Inline => self.inline_mode_assign_block_size(layout_context),
-            Direction::Block => self.block_mode_assign_block_size(),
+            Direction::Inline => {
+                self.inline_mode_assign_block_size(layout_context);
+                let block_start = AdjoiningMargins::from_margin(self.block_flow.fragment.margin.block_start);
```

Comment:
Should the code inside this branch be moved into `inline_mode_assign_block_size`?

Reply:
No, I believe they should be shared for both containers, only separated for now because column flexbox is not implemented.

Reply:
Okay, that makes sense.